pkg/dockerregistry/server/auth.go

```diff
@@ -535,3 +545,11 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func isMetricsBearerToken(ctx context.Context, token string) bool {
+	config := ConfigurationFrom(ctx)
+	if config.Metrics.Enabled {
+		return config.Metrics.Secret == token
+	}
```

Comment:
If we're already doing auth from the registry, we should use the SAR check instead of a secret.  That allows us to then use the same policy the apiservers do.